bw2data/backends/utils.py

```diff
@@ -71,9 +73,14 @@ def dict_as_activitydataset(ds):
         "product": ds.get("reference product"),
         "type": ds.get("type", labels.process_node_default),
     }
+    # Use during `insert_many` calls as these skip auto id generation because they don't call
+    # `.save()`
+    if add_snowflake_id:
+        val['id'] = next(snowflake_id_generator)
+    return val
 
 
-def dict_as_exchangedataset(ds):
+def _dict_as_exchangedataset(ds: Any) -> dict:
```

Comment:
Even though `add_snowflake_id` is optional, this can break #174, as it expects `dict_as_…` to have identical signatures. Will we ever need an `insert_many` via the `revisions` module?

Reply:
> Will we ever need an insert_many via the revisions module?
Not in our current conception as each changeset is for one object at a time, and we apply them sequentially. But I think we will need to have custom behaviour for all `peewee` methods which change state on the database, and handle each one of them.